src/Foundation/NSDate.cs

```diff
@@ -35,7 +35,17 @@
 namespace Foundation {
 
 	public partial class NSDate {
-		const long NSDATE_TICKS = 631139040000000000;
+		const double NANOSECS_PER_MILLISEC = 1000000.0;
+
+		private static readonly NSCalendar calendar;
+		private static readonly NSDateComponents components;
+
+		static NSDate()
+		{
+			components = new NSDateComponents();
```

Comment:
Can we express these as field initializers?

https://stackoverflow.com/a/610837

Reply:
If I understand correctly, updating to field initializers would mean initializing `calendar` and `components` in their declaration and removing the static ctor?

Reply:
A type without an explicit static constructor is marked as `beforefieldinit` in IL, which gives the runtime a bit more freedom into how the type is initialized, thus leading to slightly better performance.

```csharp
static readonly NSDateComponents components = new NSDateComponents();
static readonly NSCalendar calendar = new NSCalendar(NSCalendarType.Gregorian) {
    Timezone = NSTimeZone.FromName ("UTC"),
}
```

Reply:
> If I understand correctly, updating to field initializers would mean initializing calendar and components in their declaration and removing the static ctor?

Yep!

Reply:
Hmmm but the performance is when we are using the JIT, which makes sense, since the JIT will need to do extra work at run time. We are dealing with a AOT compilation, I would expect that the AOT will do the right thing and just call the static constructor when the is required. 

My gut feeling, and I have to look closer, is that if we use the static variables we will be paying a memory increase (tiny, I know) and we will be loosing laziness, while the static constructor's will only kick in when needed. 

Either way, I think we might be getting into too much trouble for NSCalendar and a NSComponents that will be needed when casting. We are improving from having to create them in the casts. So static field or static constructor. Either way. We are not dealing with a singleton, so it does not really matter.

Reply:
Not using an explicit static constructor is a win even when using the AOT compiler.

Reply:
@mandel-macaque and we still need to consider VSMac - which is JIT.

Reply:
`calendar` can be lazily initialized. Worse case is that a 2nd instance will be created and the first one eventually GC'ed.